pandas/core/indexes/range.py

```diff
@@ -152,20 +150,14 @@ def _constructor(self):
         """ return the class to use for construction """
         return Int64Index
 
-    @property
+    @cache_readonly
     def _data(self):
         """
         An int array that for performance reasons is created only when needed.
 
-        The constructed array is saved in ``_cached_data``. This allows us to
-        check if the array has been created without accessing ``_data`` and
-        triggering the construction.
```

Comment:
do we ever do this check outside of tests?

Reply:
No, and the checks in the tests are to ensure this array isn't created unless it's needed.